test/functional/test_framework/wallet.py

```diff
@@ -55,25 +55,34 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), seq_num = 0xffffffff, from_node, utxo_to_spend=None, send_tx = True, default_fee = False):
         """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
         self._utxos = sorted(self._utxos, key=lambda k: k['value'])
         utxo_to_spend = utxo_to_spend or self._utxos.pop()  # Pick the largest utxo (if none provided) and hope it covers the fee
         vsize = Decimal(96)
-        send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000))
+
+        # if send_tx is set to False then calculate send_value without vsize
```

Comment:
I'm curious why `send_tx = False` should affect the fee?

Reply:
It was just a little hack that I found to get the `vsize` and `fee` assertion to work. But, I believe I found a way out! Pushed a new commit :)